### DISCUSSION TITLE
Conditional Validation using external State

### DISCUSSION THREAD

**Discussion by danielbattat** [react-hook-form/react-hook-form#3200] [Q&A] [answered]
Hello,

I am using an external state(via React.useState), which I would like use to disable field validation. 
However, the state's value changes do not appear to propagate into the validation callback, and it remains at its initial value (I tried console.logging in the validation callback to verify this).

In this example, changing `validate` does not change within the `isDate` callback, and therefore the validation always occurs.

```
const [validate, setValidate] = useState(true);

return (
            <Controller
              as={<DatePicker />}
              name={`dueDate`}
              control={control}
              rules={{
                validate: {
                    isDate: value => !validate || value instanceof dayjs || 'Select a due date.',
                }                  
              }}
            />
);
```

I also tried using React.useMemo and React.useCallback with the validate state as a dependency and that did not work either.

Any advice on how to conditionally validate would be appreciated.

UPDATE:
CSB: https://codesandbox.io/s/react-hook-form-basic-forked-sdjri?file=/src/index.js

This only happens with controller, not with register()

**Comment by bluebill1049** [comment]
can you provide a codesandbox?

**Reply by danielbattat** [reply]
Sure-- https://codesandbox.io/s/react-hook-form-basic-forked-sdjri?file=/src/index.js

Interestingly, "amount" updates if you use register instead of Controller

**Comment by danielbattat** [comment, marked as answer]
Upgrading to v6 resolves this
